ios/FluentUI/People Picker/ContactCollectionView.swift

```diff
@@ -0,0 +1,139 @@
+//
+//  Copyright (c) Microsoft Corporation. All rights reserved.
+//  Licensed under the MIT License.
+//
+
+import UIKit
+
+@objc(MSFContactCollectionView)
+open class ContactCollectionView: UICollectionView {
+    @objc public init(personaData: [PersonaData] = []) {
+        layout = ContactCollectionViewLayout()
+        layout.scrollDirection = .horizontal
+        super.init(frame: .zero, collectionViewLayout: layout)
+
+        if personaData.count > 0 {
+            contactList = personaData
+        }
+
+        configureCollectionView()
+        setupHeightConstraint()
+        register(ContactCollectionViewCell.self, forCellWithReuseIdentifier: ContactCollectionViewCell.identifier)
+        NotificationCenter.default.addObserver(self, selector: #selector(setupHeightConstraint), name: UIContentSizeCategory.didChangeNotification, object: nil)
+    }
+
+    @objc public required init?(coder: NSCoder) {
+        preconditionFailure("init(coder:) has not been implemented")
+    }
+
+    // The array of PersonaData which is used to create each ContactView. The height constraint of the ContactCollectionView is updated when the count increases from 0 or decreases to 0.
+    @objc public var contactList: [PersonaData] = [] {
+        didSet {
+            if (oldValue.count == 0 && contactList.count > 0) || (oldValue.count > 0 && contactList.count == 0) {
+                setupHeightConstraint()
+            }
+        }
+    }
+
+    struct Constants {
+        static let leadingInset: CGFloat = 16.0
+        static let extraSmallContentContactHeight: CGFloat = 115.0
+        static let smallContentContactHeight: CGFloat = 117.0
+        static let mediumContentContactHeight: CGFloat = 118.0
+        static let largeContentContactHeight: CGFloat = 121.0
+        static let extraLargeContentContactHeight: CGFloat = 125.0
+        static let extraExtraLargeContentContactHeight: CGFloat = 129.0
+        static let extraExtraExtraLargeContentContactHeight: CGFloat = 135.0
+    }
+
+    let layout: ContactCollectionViewLayout
+    var widthConstraint: NSLayoutConstraint?
+    lazy var heightConstraint: NSLayoutConstraint = {
+        let heightConstraint = heightAnchor.constraint(equalToConstant: 0.0)
+        heightConstraint.isActive = true
```

Comment:
this seems bit weird that the hight constraint is activated by referencing this variable.

Reply:
I recommended this since it simplifies the logic. Height constraint will always be active. An alternative would be to activate it in the init method. Both seem fine to me.

Reply:
i think activate in initi seems clearer to me